backend/chunker/chunker_internal_test.go

```diff
@@ -663,6 +663,93 @@ func testMetadataInput(t *testing.T, f *Fs) {
 	runSubtest(futureMeta, "future")
 }
 
+// The newer method of doing transactions without renaming should still be able to correctly process chunks that were created with renaming
+// If you attempt to do the inverse, however, the data chunks will be ignored causing commands to perform incorrectly
+func testBackwardsCompatibility(t *testing.T, f *Fs) {
+	if !f.useMeta {
+		t.Skip("Can't do norename transactions without metadata")
+	}
+	const dir = "backcomp"
+	ctx := context.Background()
+	saveOpt := f.opt
+	saveUseRename := f.useRename
+	defer func() {
+		f.opt.FailHard = false
+		_ = operations.Purge(ctx, f.base, dir)
+		f.opt = saveOpt
+		f.useRename = saveUseRename
+	}()
+	f.opt.ChunkSize = fs.SizeSuffix(10)
+
+	modTime := fstest.Time("2001-02-03T04:05:06.499999999Z")
+	contents := random.String(250)
+	newFile := func(f fs.Fs, name string) (fs.Object, string) {
+		filename := path.Join(dir, name)
+		item := fstest.Item{Path: filename, ModTime: modTime}
+		_, obj := fstests.PutTestContents(ctx, t, f, &item, contents, true)
+		require.NotNil(t, obj)
+		return obj, filename
+	}
+
+	f.opt.FailHard = false
+	f.useRename = true
+	file, fileName := newFile(f, "renamefile")
+
+	f.opt.FailHard = false
+	item := fstest.NewItem(fileName, contents, modTime)
+
+	var items []fstest.Item
+	items = append(items, item)
+
+	f.useRename = true
+	fstest.CheckListingWithRoot(t, f, dir, items, nil, f.Precision())
+	_, err := f.NewObject(ctx, fileName)
+	assert.NoError(t, err)
+
+	f.opt.FailHard = true
+	_, err = f.List(ctx, dir)
+	assert.NoError(t, err)
+
+	f.opt.FailHard = false
+	_ = file.Remove(ctx)
+}
+
+func testServerSideMove(t *testing.T, f *Fs) {
+	if !f.useMeta {
+		t.Skip("Can't do norename transactions without metadata")
+	}
+	ctx := context.Background()
+
+	f1, err := NewFs(ctx, "x:", f.base.String(), f.m)
+	c1 := f1.(*Fs)
+	c1.useRename = true
+	c1.opt.ChunkSize = fs.SizeSuffix(3)
+	assert.NoError(t, err)
+
+	f2, err := NewFs(ctx, "y:", f.base.String(), f.m)
+	c2 := f2.(*Fs)
```

Comment:
> I added the `m` field to the chunker `Fs` struct to preserve the `configmap.Mapper` that is passed in to `NewFs`. I saw that some other backends did this for their `Fs` struct ([example](https://github.com/rclone/rclone/blob/50cf97fc729ee305cb89260f2c7ed387cf12bac0/backend/sugarsync/sugarsync.go#L194)) and it was the only way I could figure out how to access the `configmap.Mapper` from within

@ncw 
We need two ad-hoc remotes for a server-side test.
I recall you recently introduced a context attached config.
Can we use that instead of an otherwise unused link to config mapper inside the Fs struct.

Reply:
@ncw ?

Reply:
@ncw here is comment by author --> https://github.com/rclone/rclone/pull/4675#issuecomment-756367287

Reply:
I think, if we use `fs.NewFs` instead of `chunker.NewFs`, we can avoid the need for the extra mapper reference.

Reply:
My commit https://github.com/rclone/rclone/pull/4675/commits/b3b97aa8725576c79c06ddab703e7ea6c8f5775d closes this thread.